File: backend/app/schemas/owner_dashboard.py
from datetime import date
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel, ConfigDict


class OwnerDashboardMonthOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    month: date
    total_bookings: int
    revenue: Decimal
    pending: int
    accepted: int
    completed: int
    cancelled: int
    expired: int


class OwnerDashboardSummaryOut(BaseModel):
    total_bookings: int
    revenue: Decimal
    pending: int
    accepted: int
    completed: int
    cancelled: int
    expired: int


class OwnerDashboardStatsOut(BaseModel):
    period: Literal['monthly']
    from_date: date | None = None
    to_date: date | None = None
    summary: OwnerDashboardSummaryOut
    months: list[OwnerDashboardMonthOut]

File: frontend/src/api/ownerDashboard.ts
import { api } from './client'

export interface OwnerDashboardMonth {
  month: string
  total_bookings: number
  revenue: string
  pending: number
  accepted: number
  completed: number
  cancelled: number
  expired: number
}

export interface OwnerDashboardSummary {
  total_bookings: number
  revenue: string
  pending: number
  accepted: number
  completed: number
  cancelled: number
  expired: number
}

export interface OwnerDashboardStats {
  period: 'monthly'
  from_date?: string | null
  to_date?: string | null
  summary: OwnerDashboardSummary
  months: OwnerDashboardMonth[]
}

export function fetchOwnerDashboardStats(params: { from?: string; to?: string } = {}) {
  const search = new URLSearchParams()
  search.set('period', 'monthly')
  if (params.from) search.set('from', params.from)
  if (params.to) search.set('to', params.to)
  const query = search.toString()
  return api<OwnerDashboardStats>(`/owner/stats${query ? `?${query}` : ''}`)
}